### PR TITLE
Max_Sum_Rectangle-matrix Code in C

### PR DESCRIPTION
**Pull request by nutanaarohi123**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
#937 
C code to find the maximum sum subarray in the 2-D matrix using Kadane’s algorithm to reduce the time complexity to O(n^3).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#937

### CODE REVIEW

**Comment by nutanaarohi123** [comment]
Kindly review this. @SKAUL05  @iamrajiv  @Kajol-Kumari

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@nutanaarohi123 Update time complexity `O(n^3)` to `O(cols^2 * rows)` and space complexity `O(n)` to `O(rows)`.

**Comment by nutanaarohi123** [comment]
> @nutanaarohi123 Update time complexity `O(n^3)` to `O(cols^2 * rows)` and space complexity `O(n)` to `O(rows)`.

Updated the changes. Kindly review @iamrajiv.

**Review by SKAUL05 — APPROVED** [review]

File: C/dp/Max_Sum_Rectangle.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ 1 2 -1 -4 -20
+ -8 -3 4 2 1
+ 3 8 10 1 3
+ -4 -1 1 7 -6
+ 
+ Output-->
+ Maximum sumRectangle is : 29
+ 
+ Time Complexity: O(cols^2 * rows) 
+ Space Compleity: O(rows)
```
Comment (iamrajiv) [inline C/dp/Max_Sum_Rectangle.c]: @nutanaarohi123 fix typo

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C/dp/Max_Sum_Rectangle.c (hunk 1 above)
Comment (nutanaarohi123) [reply C/dp/Max_Sum_Rectangle.c]: > @nutanaarohi123 fix typo

DONE

**Review by iamrajiv — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1:

**Pull request merged**